### ISSUE TITLE
Do not return Generics with Generics with Generics

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#541]
We should report a warning if a method/property receives or returns a generic type with a generic type as type argument and a generic type as that's type argument as well.

Following code should trigger a warning:

```c#
IReadOnlyCollection<KeyValuePair<Tuple<Guid, Version>, int>> ReturnSomething();

void ReceiveSomething(IEnumerable<KeyValuePair<Tuple<Guid, Version>, int>> list);
```

It's hard to find out what's meant, in addition it seems like there is some type information hiding within the generics due to primitive obsession.

It's much easier to see what is meant if concrete types would be used instead of generic type arguments.
